validation/datehelpers_test.go

```diff
@@ -0,0 +1,27 @@
+package validation
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLongDuration(t *testing.T) {
+	var now = time.Now()
+	t.Run("Empty duration", func(t *testing.T) {
+		assert.False(t, IsValidLongDuration(""))
+		assert.Equal(t, now, AddLongDuration(now, ""))
+	})
+	t.Run("Invalid duration", func(t *testing.T) {
+		var duration = "2d3y4"
+		assert.False(t, IsValidLongDuration(duration))
+	})
+	t.Run("Valid duration", func(t *testing.T) {
+		var duration = "2d3y4w1m"
+		assert.True(t, IsValidLongDuration(duration))
+
+		var after = now.AddDate(3, 1, 4*7+2)
+		assert.Equal(t, after, AddLongDuration(now, duration))
```

Comment:
What happens if we have 4y3w2m2w ?

Reply:
As you can see further, values are summed.